pkg/autoupdate/tuf/autoupdate.go

```diff
@@ -287,11 +320,30 @@ func (ta *TufAutoupdater) checkForUpdate() error {
 		return fmt.Errorf("could not download updates: %+v", updateErrors)
 	}
 
-	for _, updateDownloaded := range updatesDownloaded {
-		if updateDownloaded {
-			// In the future, we would restart or re-launch the binary with the new version
-			level.Debug(ta.logger).Log("msg", "at least one update downloaded")
-			break
+	// If launcher was updated, we want to exit and reload
+	if updatedVersion, ok := updatesDownloaded[binaryLauncher]; ok {
+		level.Debug(ta.logger).Log("msg", "launcher updated -- exiting to load new version", "new_binary_version", updatedVersion)
+		ta.signalRestart <- NewLauncherReloadNeededErr(updatedVersion)
+		return nil
+	}
+
+	// For non-launcher binaries (i.e. osqueryd), call any reload functions we have saved
+	for binary, newBinaryVersion := range updatesDownloaded {
+		if binary == binaryLauncher {
+			continue
+		}
+
+		if restart, ok := ta.restartFuncs[binary]; ok {
+			if err := restart(); err != nil {
+				level.Debug(ta.logger).Log("msg", "failed to restart binary after update",
+					"binary", binary,
+					"new_binary_version", newBinaryVersion,
+					"err", err)
+			} else {
+				level.Debug(ta.logger).Log("msg", "restarted binary after update",
+					"binary", binary,
+					"new_binary_version", newBinaryVersion)
+			}
```

Comment:
```suggestion
		if restart, ok := ta.restartFuncs[binary]; ok {
			if err := restart(); err != nil {
				level.Debug(ta.logger).Log("msg", "failed to restart binary after update",
					"binary", binary,
					"new_binary_version", newBinaryVersion,
					"err", err)
					continue
			}
		  level.Debug(ta.logger).Log("msg", "restarted binary after update",
			  "binary", binary,
			  "new_binary_version", newBinaryVersion)	
```

Reply:
Someday I will submit a PR where you don't have to make this suggestion 😅 (I promise I do look for these but sometimes miss them during refactors) -- I'll update!